### PR TITLE
Bugfix for undefined threshold from Scrublet

### PR DESCRIPTION
**Pull request by pinin4fjords**
This is a bugfix to allow for the fact that Scrublet doesn't always find a threshold and produce predictions by:

   - Detecting when threshold not present, and setting prediction column to False'
   - Not assuming presence of threshold in plot functions.

### CODE REVIEW

File: scanpy/external/pl.py [hunk 1]
```diff
@@ -372,7 +372,11 @@ def scrublet_score_distribution(
         simulation separately for advanced usage.
     """
 
-    threshold = adata.uns['scrublet']['threshold']
+    threshold = None
+
+    if 'threshold' in adata.uns['scrublet']:
+        threshold = adata.uns['scrublet']['threshold']
```
Comment (ivirshup) [inline scanpy/external/pl.py]: A bit shorter:

```suggestion
    threshold = adata.uns["scrublet"].get("threshold", None)
```

**Review by ivirshup — COMMENTED** [review]
Thanks for the fix. I think these approaches make sense.

Could this get tests? I'm thinking

* Make a case where a threshold can't be found (not sure how this would be done)
  * scrublet warns the user
  * Check plots don't error
  * Check that nothing is predicted as a doublet

**Comment by pinin4fjords** [comment]
> * Make a case where a threshold can't be found (not sure how this would be done)

Obviously I can test the plotting by directly unsetting the relevant things, but I'm not actually sure how to trigger a failure with the test data I'm afraid.

**Review by Zethson — APPROVED** [review]
LGTM

**Pull request merged**